api/tests/unit/features/test_unit_feature_external_resources_views.py

```diff
@@ -0,0 +1,316 @@
+import datetime
+
+import pytest
+import simplejson as json
+from django.core.serializers.json import DjangoJSONEncoder
+from django.urls import reverse
+from freezegun import freeze_time
+from pytest_lazyfixture import lazy_fixture
+from rest_framework import status
+from rest_framework.test import APIClient
+
+from features.feature_external_resources.models import FeatureExternalResource
+from features.models import Feature
+from integrations.github.models import GithubConfiguration, GithubRepository
+from projects.models import Project
+
+_django_json_encoder_default = DjangoJSONEncoder().default
+
+
+def mocked_requests_post(*args, **kwargs):
+    class MockResponse:
+        def __init__(self, json_data, status_code):
+            self.json_data = json_data
+            self.status_code = status_code
+
+        def raise_for_status(self) -> None:
+            pass
+
+        def json(self):
+            return self.json_data
+
+    return MockResponse(json_data={"data": "data"}, status_code=200)
+
+
+@pytest.mark.parametrize(
+    "client",
+    [lazy_fixture("admin_master_api_key_client"), lazy_fixture("admin_client")],
+)
+@freeze_time("2024-01-01")
+def test_create_feature_external_resource(
+    client: APIClient,
+    feature_with_value: Feature,
+    project: Project,
+    github_configuration: GithubConfiguration,
+    github_repository: GithubRepository,
+    mocker,
+) -> None:
+    # Given
+    mock_generate_token = mocker.patch(
+        "integrations.github.github.generate_token",
+    )
+    mock_generate_token.return_value = "mocked_token"
+    github_request_mock = mocker.patch(
+        "requests.post", side_effect=mocked_requests_post
+    )
+    datetime_now = datetime.datetime.now()
+
+    feature_external_resource_data = {
+        "type": "GITHUB_ISSUE",
+        "url": "https://github.com/repoowner/repo-name/issues/35",
+        "feature": feature_with_value.id,
+        "metadata": {"status": "open"},
+    }
+
+    url = reverse(
+        "api-v1:projects:feature-external-resources-list",
+        kwargs={"project_pk": project.id, "feature_pk": feature_with_value.id},
+    )
+
+    # When
+    response = client.post(url, data=feature_external_resource_data, format="json")
+
+    github_request_mock.assert_called_with(
+        "https://api.github.com/repos/repoowner/repo-name/issues/35/comments",
+        json={
+            "body": f"### This pull request is linked to a Flagsmith Feature (feature_with_value):\n**Test Environment**\n- [ ] Disabled\nunicode\n```value```\n\nLast Updated {datetime_now.strftime('%dth %b %Y %I:%M%p')}"  # noqa E501
+        },
+        headers={
+            "Accept": "application/vnd.github.v3+json",
+            "Authorization": "Bearer mocked_token",
+        },
+        timeout=10,
+    )
+    # Then
+    assert response.status_code == status.HTTP_201_CREATED
+    # assert that the payload has been save to the database
+    db_record = FeatureExternalResource.objects.filter(
```

Comment:
nit: can we rename this variable

Reply:
Renamed to `feature_external_resources`.